tests/functional/conftest.py

```diff
@@ -401,6 +401,21 @@ def _merge_request(*, source_branch: str):
                 "commit_message": "New commit in new branch",
             }
         )
+        if create_pipeline:
+            project.files.create(
+                {
+                    "file_path": ".gitlab-ci.yml",
+                    "branch": source_branch,
+                    "content": """
+test:
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event"'
+  script:
+    - sleep 24h  # We don't expect this to finish
+""",
+                    "commit_message": "Add a simple pipeline",
+                }
+            )
```

Comment:
Does it make sense to move this block into a fixture and re-use it in other functions?

Reply:
@lmilbaum I think this makes sense if we use plain fixtures but this one is meant to be used as a function inside the test. I've created https://github.com/python-gitlab/python-gitlab/issues/2353 to tackle this for all fixtures consistently.